include/libopencm3/pac55xx/can.h

```diff
@@ -0,0 +1,289 @@
+ * @{*/
+/** This is the 32-bit memory mapped read/write accessor for:
+ * - BTR1 - bits 31:24 - Bus Timing 1 Register RW, default 00h
+ * - BTR0 - bits 23:16 - Bus Timing 0 Register RW, default 00h
+ * - RMC  - bits 15:8  - Receive Message Counter RO, default 00h
+ * - IMR  - bits  7:0  - Interrupt Mask Register RW, default 00h
+ */
+#define CAN_BTR1_BTR0_RMC_IMR(can_base)         MMIO32((can_base) + 0x0004)
+/**@}*/
+
+/** CAN Transmit Buffer Register RW, default 00000000h */
+#define CAN_TXBUF(can_base)     MMIO32((can_base) + 0x0008)
+/** CAN Receive Buffer Register RO, default 00000000h */
+#define CAN_RXBUF(can_base)     MMIO32((can_base) + 0x000C)
+/** CAN Acceptance Code Register RW, default 00000000h */
+#define CAN_ACR(can_base)       MMIO32((can_base) + 0x0010)
+/** CAN Acceptance Mask Register RW, default 00000000h */
+#define CAN_AMR(can_base)       MMIO32((can_base) + 0x0014)
+
+/**
+ * @defgroup can_alc_txrxerr_ecc CAN ALC/TXERR/RXERR/ECC Registers
+ * @{*/
+#define CAN_ALC_TXERR_RXERR_ECC(can_base)   MMIO32((can_base) + 0x0018)
+/** CAN Error Code Capture Register RO, default 00h */
+#define CAN_ECC(can_base)       (CAN_ALC_TXERR_RXERR_ECC(can_base) & 0xFF)
+/** CAN RX Error Counter Register RO, default 00h */
+#define CAN_RXERR(can_base)     ((CAN_ALC_TXERR_RXERR_ECC(can_base) >> 8) & 0xFF)
+/** CAN TX Error Counter Register RO, default 00h */
+#define CAN_TXERR(can_base)     ((CAN_ALC_TXERR_RXERR_ECC(can_base) >> 16) & 0xFF)
+/** CAN Arbitration Lost Code Capture Register RO, default 00h */
+#define CAN_ALC(can_base)       ((CAN_ALC_TXERR_RXERR_ECC(can_base) >> 24) & 0xFF)
+/**@}*/
+
+/** CAN Mode Register bit definitions. This register controls high level modes of the CAN peripheral.
+ * @defgroup can_mr_bits CAN Mode Register
+ * @{*/
+/** AFM: Acceptance Filter Mode */
+#define CAN_MR_AFM      BIT0
+/** LOM: Listen only mode */
+#define CAN_MR_LOM      BIT1
+/** RM: Reset Mode */
+#define CAN_MR_RM       BIT2
+/**@}*/
+
+/** CAN Command Register. This register commands the CAN peripheral to either transmit or abort.
+ * @defgroup can_cmr_bits CAN Command Register
+ * @{*/
+/** AT: Abort transmission */
+#define CAN_CMR_AT      BIT9
+/** TR: Transmit Request */
+#define CAN_CMR_TR      BIT10
+/**@}*/
+
+/** CAN Status Register. This register provides read-only status of the CAN peripheral.
+ * @defgroup can_sr_bits CAN Status Register
+ * @{*/
+/** BS: Bus Off Status */
+#define CAN_SR_BS       BIT16
+/** ES: Error Status */
+#define CAN_SR_ES       BIT17
+/** TS: Transmit Status */
+#define CAN_SR_TS       BIT18
+/** RS: Receive Status */
+#define CAN_SR_RS       BIT19
+/** TBS: Transmit Buffer Status */
+#define CAN_SR_TBS      BIT21
+/** DSO: Data Overrun Status */
+#define CAN_SR_DSO      BIT22
+/** RBS: Receive Buffer Status */
+#define CAN_SR_RBS      BIT23
+/**@}*/
+
+/** CAN Interrupt Status Register bit definitions.
+ * 1: interrupt cleared
+ * 0: interrupt triggered
```

Comment:
This didn't look write, and doesn't match the docs either. it should be the other way around

Reply:
Looks like I need to fix this documentation.  The bit goes high when the peripheral triggers the interrupt.  However, in order to clear the interrupt (by the user), you have to write a '1' to the bit.